### PR TITLE
test: port lock table ut from mysql

### PR DESCRIPTION
**Pull request by marsishandsome**
### What problem does this PR solve? 
port some unit test from mysql for lock table (https://github.com/pingcap/tidb/pull/10343)

### What is changed and how it works?
add some unit test

Tests 

 - Unit test

### CODE REVIEW

**Comment by marsishandsome** [comment]
@razycs520 PTAL

**Comment by marsishandsome** [comment]
@winkyao PTAL

**Comment by marsishandsome** [comment]
@winkyao PTAL

File: ddl/db_test.go [hunk 1]
```diff
@@ -2990,6 +2990,141 @@ func (s *testDBSuite4) TestAlterShardRowIDBits(c *C) {
...
+	tk.MustExec("CREATE TABLE t2 (a int)")
+	tk.MustExec("lock tables t1 write, t2 write")
+	tk.MustExec("drop table t1,t2")
+
+	tk.MustExec("CREATE TABLE t1 (a int)")
+	tk.MustExec("CREATE TABLE t2 (a int)")
+	tk.MustExec("lock tables t1 write, t2 write")
+	tk.MustExec("drop table t2,t1")
+
+	tk.MustExec("drop table if exists t1,t2")
```
Comment (crazycs520) [inline ddl/db_test.go]: redundant?

File: ddl/db_test.go [hunk 2]
```diff
@@ -2990,6 +2990,141 @@ func (s *testDBSuite4) TestAlterShardRowIDBits(c *C) {
...
+	_, err := tk2.Exec("SELECT * FROM t1")
+	c.Assert(terror.ErrorEqual(err, infoschema.ErrTableLocked), IsTrue)
+
+	tk.Se.Close()
+
+	tk2.MustExec("SELECT * FROM t1")
+	tk2.MustExec("DROP TABLE t1")
+
+	tk.MustExec("unlock tables")
+	tk2.MustExec("DROP TABLE IF EXISTS t1")
```
Comment (crazycs520) [inline ddl/db_test.go]: redundant?

File: ddl/db_test.go (hunk 1 above)
Comment (marsishandsome) [reply ddl/db_test.go]: fixed

File: ddl/db_test.go (hunk 2 above)
Comment (marsishandsome) [reply ddl/db_test.go]: fixed

**Comment by marsishandsome** [comment]
@crazycs520 PTAL again

**Comment by zz-jason** [comment]
@marsishandsome thanks for your contribution, please follow [CONTRIBUTING.md](https://github.com/pingcap/tidb/blob/master/CONTRIBUTING.md) to refine the PR title.

**Comment by zz-jason** [comment]
@crazycs520 The two `LGTM` are all given by you, I think we need another `LGTM` from another different reviewer.

**Comment by crazycs520** [comment]
@tangenta PTAL

File: ddl/db_test.go [hunk 3]
```diff
@@ -2990,6 +2990,138 @@ func (s *testDBSuite4) TestAlterShardRowIDBits(c *C) {
...
+	tk := s.tk
+	tk.MustExec("use test")
+	tk2.MustExec("use test")
+	tk.MustExec("drop table if exists t1")
+	tk.MustExec("create table t1 ( n int auto_increment primary key)")
+
+	// Test: allow read
+	tk.MustExec("lock tables t1 write local")
+	tk.MustExec("insert into t1 values(NULL)")
+	tk2.MustQuery("select count(*) from t1")
```
Comment (tangenta) [inline ddl/db_test.go]: Do we need to check it's result?

**Comment by marsishandsome** [comment]
> @marsishandsome thanks for your contribution, please follow [CONTRIBUTING.md](https://github.com/pingcap/tidb/blob/master/CONTRIBUTING.md) to refine the PR title.

done

File: ddl/db_test.go (hunk 3 above)
Comment (marsishandsome) [reply ddl/db_test.go]: we do not need to check the result (see mysql `lock table` test).

**Pull request merged**